esmvaltool/cmorizers/data/utilities.py

```diff
@@ -455,36 +467,33 @@ def fix_dim_coordnames(cube):
                 coord_type)
             continue
 
-        if coord_type == 'T':
-            coord.var_name = 'time'
-            coord.attributes = {}
-
-        if coord_type == 'X':
-            coord.var_name = 'lon'
-            coord.standard_name = 'longitude'
-            coord.long_name = 'longitude coordinate'
-            coord.units = Unit('degrees')
-            coord.attributes = {}
-
-        if coord_type == 'Y':
-            coord.var_name = 'lat'
-            coord.standard_name = 'latitude'
-            coord.long_name = 'latitude coordinate'
-            coord.units = Unit('degrees')
-            coord.attributes = {}
-
-        if coord_type == 'Z':
-            if coord.var_name == 'depth':
-                coord.standard_name = 'depth'
-                coord.long_name = \
-                    'ocean depth coordinate'
-                coord.var_name = 'lev'
-                coord.attributes['positive'] = 'down'
-            if coord.var_name == 'pressure':
-                coord.standard_name = 'air_pressure'
-                coord.long_name = 'pressure'
-                coord.var_name = 'air_pressure'
-                coord.attributes['positive'] = 'up'
+        coord.attributes = {}
+
+        # lookup table
+        lut = {
+            'T': 'time',
+            'X': 'longitude',
+            'Y': 'latitude',
+            'Z': {
+                'depth': ('depth_coord', 'down'),
+                # Any pressure coord would work.
+                'pressure': ('p220', 'up')
+                }}
+
+        match = lut.get(coord_type)
+        tindex = match if isinstance(match, str) else None
+        if isinstance(match, dict):
+            nested = match.get(coord.var_name)
+            tindex, coord.attributes['positive'] = (
+                nested if nested else (None, None))
+
+        if tindex:
+            coord_info = CMOR_TABLES[project_id].coords[tindex]
+            coord.var_name = coord_info.out_name
+            coord.standard_name = coord_info.standard_name
+            coord.long_name = coord_info.long_name
+            if tindex != 'time':
```

Comment:
Notice that units are not being set either for `Z` coordinates. Could you just set them for lat and lon?

Reply:
Updated so that it only sets units for lat/lon.